annotation-file-utilities/src/annotator/find/ASTPathCriterion.java

```diff
@@ -140,7 +135,7 @@ public boolean isSatisfiedBy(TreePath path) {
 
         int astPathLen = astPath.size();
         int actualPathLen = actualPath.size();
-        if (astPathLen == 0 || actualPathLen == 0) { return false; }
+        if (actualPathLen <= 0) { return false; }
```

Comment:
True, in the current code `actualPathLen` cannot be negative, so `==` would work.  However,`<=` more precisely expresses the required invariant -- that only a positive integer is valid here -- independently from the size() invariant, so I find it preferable.  (If you strongly disagree, I will change it.)